editor/components/block-list/layout.js

```diff
@@ -228,42 +230,51 @@ class BlockListLayout extends Component {
 						layout={ defaultLayout }
 						isFirst={ blockIndex === 0 }
 						isLast={ blockIndex === blockUIDs.length - 1 }
+						isLocked={ isLocked }
 						renderBlockMenu={ renderBlockMenu }
 					/>
 				) ) }
-				<DefaultBlockAppender
+				{ ! isLocked && <DefaultBlockAppender
 					rootUID={ rootUID }
 					lastBlockUID={ last( blockUIDs ) }
 					layout={ defaultLayout }
-				/>
+				/> }
```

Comment:
Any particular reason why you've used `isLocked` here instead of doing it inside `DefaultBlockAppender` (I guess it's the same question for passing it as prop to `Block`)?

I tend to prefer components to be autonomous as much as we can

Reply:
Hi @youknowriad, I was doing some explorations (when I found this bug) related to adding a nested button to cover image, and in that case, even if we use nesting that nesting is locked, it contains just a button we cannot add new blocks etc. So the logic for isLocked will be more complex and doing it here would avoid repeating it. That said, for now, that is not still the case so I agree with you and later if the situation changes we can change the code to the new logic.

Reply:
It was updated :)